cmd/buildah/common.go

```diff
@@ -1,15 +1,28 @@
 package main
 
 import (
+	"encoding/json"
 	"os"
+	"strings"
+	"time"
 
 	is "github.com/containers/image/storage"
+	"github.com/containers/image/types"
 	"github.com/containers/storage"
 	"github.com/pkg/errors"
 	"github.com/projectatomic/buildah"
 	"github.com/urfave/cli"
 )
 
+type imageMetadata struct {
+	Tag            string              `json:"tag"`
+	CreatedTime    time.Time           `json:"created-time"`
+	ID             string              `json:"id"`
+	Blobs          []types.BlobInfo    `json:"blob-list"`
+	Layers         map[string][]string `json:"layers"`
+	SignatureSizes []string            `json:"signature-sizes"`
+}
```

Comment:
I wouldn't recommend parsing the metadata that the image library records - it intentionally doesn't make that a public type, so that it can change the layout to fit its needs.  In the future, it very likely will be missing the information we're trying to read, since it will be duplicating information that newer versions of the storage library will cache in publicly-defined structures.